### PR TITLE
feat: make requester an enum.

### PR DESCRIPTION
**Pull request by gdethier**
logion-network/logion-internal#297

### CODE REVIEW

File: pallets/logion_loc/src/lib.rs [hunk 1]
```diff
@@ -200,13 +224,38 @@ pub mod pallet {
 				};
 				<LocMap<T>>::insert(loc_id, loc);
```
Comment (benoitdevos) [inline pallets/logion_loc/src/lib.rs:225]: I would rather do the insert at the end of the function, just before emitting the event. If you keep it like this, you have to ensure in your unit test that this line is correctly rollbacked.

**Review by benoitdevos — APPROVED** [review]
Good job! One small remark.

File: pallets/logion_loc/src/lib.rs [hunk 2]
```diff
@@ -200,13 +224,38 @@ pub mod pallet {
...
+						Err(Error::<T>::UnexpectedRequester)?
+					},
+					RequesterOf::<T>::Loc(requester_loc_id) => if loc_type != LocType::Transaction {
+						Err(Error::<T>::UnexpectedRequester)?
+					} else {
+						let requester_loc = <LocMap<T>>::get(&requester_loc_id);
+						match requester_loc {
+							None => Err(Error::<T>::UnexpectedRequester)?,
+							Some(loc) => if loc.loc_type != LocType::Identity {
+								Err(Error::<T>::UnexpectedRequester)?
```
Comment (benoitdevos) [inline pallets/logion_loc/src/lib.rs:238]: IMHO this should also fail when identity LOC:
- is not CLOSED.
- is VOID.

**Review by benoitdevos — COMMENTED** [review]
One more remark:

File: pallets/logion_loc/src/lib.rs (hunk 2 above)
Comment (gdethier) [reply pallets/logion_loc/src/lib.rs]: Indeed!

File: pallets/logion_loc/src/lib.rs (hunk 1 above)
Comment (gdethier) [reply pallets/logion_loc/src/lib.rs:224]: Indeed, done.

**Review by benoitdevos — APPROVED** [review]
Great job !!

**Pull request merged**